### PR TITLE
1764 - migrate momentjs -> dayjs

### PR DESCRIPTION
**Pull request by flacoman91**
reduce techical debt. get off momentjs since its deprecated and dayjs  is better. 

Fixed unit tests
added some validation to convert dashes to /

### CODE REVIEW

File: src/constants/index.jsx [hunk 1]
```diff
@@ -4,6 +4,9 @@ export const MODAL_SHOWN = 'MODAL_SHOWN'
 
 export const DATE_RANGE_MIN = '2011-12-01T12:00:00.000Z'
 
+export const DATE_VALIDATION_FORMAT = 'M/D/YYYY';
```
Comment (anselmbradford) [inline src/constants/index.jsx:7]: You could add a comment above this line with a link to `https://day.js.org/docs/en/display/format` so it's clearer this is set by dayjs.

File: src/utils/__tests__/chart.spec.jsx [hunk 2]
```diff
@@ -129,7 +133,8 @@ describe( 'getTooltipTitle', () => {
     const inDate = '2020-07-01T00:00:00.000Z'
     res = sut.getTooltipTitle( inDate, interval, dateRange, true )
     expect( res ).toContain( 'Date range: 7/14/2020 - ' )
-    expect( res ).toEqual( 'Date range: 7/14/2020 - 9/30/2020' )
+    // TODO: figure out why this doesn't work in CI
+    // expect( res ).toEqual( 'Date range: 7/14/2020 - 9/30/2020' )
```
Comment (anselmbradford) [inline src/utils/__tests__/chart.spec.jsx:137]: You want to do this in another PR or this one?

Comment (flacoman91) [reply src/utils/__tests__/chart.spec.jsx:137]: I'd do this in another one. as far as i  could tell, I  could never get this to work in CI even when it was using momentjs.

Comment (flacoman91) [reply src/utils/__tests__/chart.spec.jsx]: Sorry, i took another look at this.   Looks like some of the other cases  were  incorrect and i fixed up those   tests and updated the logic.

Comment (flacoman91) [reply src/utils/__tests__/chart.spec.jsx]: and it passes locally and in CI now!

**Review by anselmbradford — APPROVED** [review]
